server/src/main/java/org/opensearch/index/mapper/ConstantFieldType.java

```diff
@@ -91,7 +91,7 @@ public final Query termQuery(Object value, QueryShardContext context) {
     }
 
     @Override
-    public final Query termQueryCaseInsensitive(Object value, QueryShardContext context) {
+    public Query termQueryCaseInsensitive(Object value, QueryShardContext context) {
```

Comment:
Why do we need to remove the final?

Reply:
For constant keyword field we would match a document only when the value is an exact match. Case Insensitive matching would mean that we allow a variation of the input 
Hence In the constantKeywordField Subclass the method is being overridden and case insensitive queries are being rejected

Please do tell if this thought process does not seem right to you